src/org/traccar/protocol/NavisProtocolDecoder.java

```diff
@@ -272,41 +317,343 @@ private Object processArray(DeviceSession deviceSession, Channel channel, ByteBu
         return positions;
     }
 
-    private Object processHandshake(Channel channel, SocketAddress remoteAddress, ByteBuf buf) {
-        buf.readByte(); // colon symbol
-        if (getDeviceSession(channel, remoteAddress, buf.toString(StandardCharsets.US_ASCII)) != null) {
-            sendReply(channel, Unpooled.copiedBuffer("*<S", StandardCharsets.US_ASCII));
+    private boolean checkFlexBitfield(int index) {
+        int byteIndex = Math.floorDiv(index, 8);
+        int bitIndex = Math.floorMod(index, 8);
+        return BitUtil.check(flexBitfield[byteIndex], 7 - bitIndex);
+    }
+
+    private void skipFlexField(int index, ByteBuf buf) {
+        if (index < FLEX_FIELDS_SIZES.length) {
+            buf.skipBytes(FLEX_FIELDS_SIZES[index]);
+        }
+    }
+
+    private ParseResult parseFlexPosition(DeviceSession deviceSession, ByteBuf buf) {
+
+        Position position = new Position(getProtocolName());
+
+        position.setDeviceId(deviceSession.getDeviceId());
+
+        long index = 0;
+        int status = 0;
+        short input = 0;
+        short output = 0;
+
+        for (int i = 0; i < flexBitfieldDataSize; i++) {
+            if (!checkFlexBitfield(i)) {
+                // skip FLEX field
```

Comment:
Please remove unnecessary comments like this.

Reply:
Fixed